code/game/machinery/bots/cleanbot.dm

```diff
@@ -392,3 +397,33 @@ text("<A href='?src=\ref[src];operation=oddbutton'>[src.oddbutton ? "Yes" : "No"
 		to_chat(P.pai, "<span class='info'><b>Welcome to your new body. Remember: you're a pAI inside a cleanbot, so get cleaning.</b></span>")
 		to_chat(P.pai, "<span class='info'>- Click on something: Scrub it clean.</span>")
 		to_chat(P.pai, "<span class='info'>If you want to exit the cleanbot, somebody has to right-click you and press 'Remove pAI'.</span>")
+
+/obj/machinery/bot/cleanbot/roomba/meido
+	name = "maidbot"
+	desc = "A small, plate-like cleaning robot. It looks quite concerned. This one has a frilly headband attached to the top."
+	icon_state = "maidbot0"
+	icon_initial = "maidbot"
+	armed = 0
+
+/obj/machinery/bot/cleanbot/roomba/meido/attackby(var/obj/item/W, mob/user)
+	if(istype(W, /obj/item/weapon/kitchen/utensil/fork) || istype(W, /obj/item/weapon/lighter))
+		to_chat(user, "<span class='notice'>\the [src] buzzes and recoils at \the [W]. Perhaps it would prefer something more refined?</span>")
+		return
+	else if (istype(W, /obj/item/clothing/head/maidhat))
+		to_chat(user, "<span class='notice'>\the [src] is already wearing one of those!</span>")
+		return
+	else if(istype(W, /obj/item/weapon/kitchen/utensil/knife/large) && !armed && user.a_intent != I_HURT)
```

Comment:
Large knife variants include the thematically-cultish dagger, the kitchen cleaver and the antagonist cleaver

Reply:
destroy all oldcoders